docs(messaging, android): add note about permissions on API33+ (#6860)
invertase/react-native-firebase: docs/messaging/usage/index.md

The permissions API for iOS provides much more fine-grain control over permissions and how they're handled within your
application. To learn more, view the advanced [iOS Permissions](/messaging/ios-permissions) documentation.

> On Android pre api level 33 you do not need to request user permission. This method can still be called on Android devices; however, and will always resolve successfully. For api level 33+ you will need to request the permission manually.

Reviewer: I normally suggest people use `react-native-permissions` as it handles everything very nicely but I like the example using nothing but built-in APIs. So I added some text mentioning both, what do you think?

```suggestion
On Android pre api level 33 you do not need to request user permission. This method can still be called on Android devices; however, and will always resolve successfully. For api level 33+ you will need to request the permission manually using either the built-in react-native `PermissionsAndroid` APIs or a related module such as `react-native-permissions`
```

Author: looks good

> On Android pre api level 33 you do not need to request user permission. This method can still be called on Android devices; however, and will always resolve successfully. For api level 33+ you will need to request the permission manually using either the built-in react-native `PermissionsAndroid` APIs or a related module such as `react-native-permissions`

Reviewer: ```suggestion
On Android api level 32 and below, you do not need to request user permission. This method can still be called on Android devices; however, and will always resolve successfully. For api level 33+ you will need to request the permission manually using either the built-in react-native `PermissionsAndroid` APIs or a related module such as `react-native-permissions`
```

> On Android api level 32 and below, you do not need to request user permission. This method can still be called on Android devices; however, and will always resolve successfully. For api level 33+ you will need to request the permission manually using either the built-in react-native `PermissionsAndroid` APIs or a related module such as `react-native-permissions`

Reviewer: ```suggestion
On Android API level 32 and below, you do not need to request user permission. This method can still be called on Android devices; however, and will always resolve successfully. For API level 33+ you will need to request the permission manually using either the built-in react-native `PermissionsAndroid` APIs or a related module such as `react-native-permissions`
```
